### DISCUSSION TITLE
Account recovery - with recovery address

### DISCUSSION THREAD

**Discussion by ghost** [farcasterxyz/protocol#162] [Q&A] [answered]
How do I recover an account with an EOA set as recovery address?

https://warpcast.com/deodad/0x4e15611c

**Comment by horsefacts** [comment, marked as answer]
You'll need to generate an [EIP-712 `Transfer` signature](https://docs.farcaster.xyz/reference/contracts/reference/id-registry#transfer-signature) from the address that will receive the recovered account, then call [`IdRegistry.recover`](https://docs.farcaster.xyz/reference/contracts/reference/id-registry#recover) from the EOA you set as the recovery address.

Collecting the signature is the tricky part. See the docs linked above for an example of how to generate it. Once you've done that, you can call [recover](https://optimistic.etherscan.io/address/0x00000000fc6c5f01fc30151999387bb99a9f489b#writeContract#F11) from Etherscan.

I know this is a little low level right now. We intend to publish a frontend tool to do this at some point.

**Reply by ghost** [reply]
thank you, I will try somehow
